### DISCUSSION TITLE
NDI Issues (Stuttering on livestream)

### DISCUSSION THREAD

**Discussion by MrBen1909** [DistroAV/DistroAV#1106] [Q&A - Support]
So I have been using NDI for just over a year now and recently upgraded to the newest version of OBS and NDI6 along with the new DistroAV plugin. Since making these changes I am getting what looks to be bursts of lag in my stream. 

I have tested the network along with a variety of other things and still cant seem to find the cause of the issue. 

Upon using the NDI Analysis tool I get a lot of Audio Connection Opened and Closed from the OBS NDI Host. 

11:49:17.082: Video data rate (Mbps). Avg=57.20
11:49:17.082: Video size (KB). Min=344.77 Avg=403.17 +/- 29.67 Max=421.64
11:49:17.082: Video recv (ms). Min=2.06 Avg=57.75 +/- 104.96 Max=502.60
11:49:17.082: Video send (ms). Min=4.06 Avg=59.89 +/- 102.95 Max=516.76
11:49:17.189: Audio connection opened.
11:49:17.418: Audio connection closed.
11:49:17.569: Audio connection opened.
11:49:17.797: Audio connection closed.
11:49:17.949: Audio connection opened.
11:49:18.178: Audio connection closed.
11:49:18.329: Audio connection opened.
11:49:18.558: Audio connection closed.
11:49:18.709: Audio connection opened.
11:49:18.938: Audio connection closed.
11:49:19.089: Audio connection opened.
11:49:19.318: Audio connection closed.
11:49:19.470: Audio connection opened.
11:49:19.699: Audio connection closed.
11:49:19.849: Audio connection opened.
11:49:20.068: Audio connection closed.
11:49:20.219: Audio connection opened.
11:49:20.447: Audio connection closed.
11:49:20.599: Audio connection opened.
11:49:20.828: Audio connection closed.
11:49:20.979: Audio connection opened.
11:49:21.209: Audio connection closed.
11:49:21.359: Audio connection opened.
11:49:21.588: Audio connection closed.
11:49:21.739: Audio connection opened.
11:49:22.087: Video data rate (Mbps). Avg=14.14
11:49:22.087: Video size (KB). Min=349.67 Avg=414.04 +/- 21.96 Max=421.92
11:49:22.087: Video recv (ms). Min=2.23 Avg=240.91 +/- 138.85 Max=423.07
11:49:22.087: Video send (ms). Min=12.37 Avg=210.83 +/- 127.51 Max=418.38
11:49:27.093: Video data rate (Mbps). Avg=13.94
11:49:27.093: Video size (KB). Min=349.87 Avg=409.58 +/- 25.72 Max=421.66
11:49:27.093: Video recv (ms). Min=2.49 Avg=261.63 +/- 144.48 Max=504.68
11:49:27.093: Video send (ms). Min=32.49 Avg=305.87 +/- 114.54 Max=532.74
11:49:27.721: Audio connection closed.
11:49:27.763: Video connection closed.
11:49:29.874: Audio connection opened.
11:49:29.914: Video connection opened.
11:49:29.917: Source is using NDI 6.0.1.0 on the WIN64 platform.
11:49:30.159: Audio connection closed.
11:49:30.310: Audio connection opened.
11:49:30.538: Audio connection closed.
11:49:30.690: Audio connection opened.
11:49:30.723: Audio connection closed.
11:49:30.874: Audio connection opened.
11:49:31.099: Audio connection closed.
11:49:31.249: Audio connection opened.
11:49:31.469: Audio connection closed.
11:49:31.620: Audio connection opened.
11:49:31.848: Audio connection closed.
11:49:31.999: Audio connection opened.
11:49:32.098: Video data rate (Mbps). Avg=35.04
11:49:32.098: Video size (KB). Min=347.59 Avg=404.70 +/- 30.12 Max=421.62
11:49:32.098: Video recv (ms). Min=1.96 Avg=93.12 +/- 112.03 Max=294.44
11:49:32.098: Video send (ms). Min=16.07 Avg=59.09 +/- 83.24 Max=250.04

Does anyone have any ideas as to where I should be looking next? 

Thanks

**Comment by winterfrost1** [comment]
TLDR: Try using NDI Tools Access Manager to change the Receive protocol.  (In my case, anything except "Reliable UDP.")

Just putting this out there in case anyone else is having similar issues: after fighting with OBS/NDI with audio cutout issues (usually logged as audio lagging, and then resetting at maximum buffers--see https://github.com/DistroAV/DistroAV/issues/875) for months, and finding no help online ("it's your network" was always the conclusion... but my network, like many others, shows no signs of issues with other applications, internet and local speed tests, file transfers, games, etc.) I finally found a solution.

Running NDI Analysis on my PCs showed the same audio connection flapping that you have reported.  Immediately following the flapping, the next audio latency times were ~200ms higher than they were before.  Every time the flapping occurs, the lag increases by ~200ms and this eventually causes the connection to lag out.  I had tried multiple PCs, and went so far as to purchase a new PC and new core switch, but all systems (Windows 11) on both switches had the same issue.

Seeing the connection flapping suggested an unreliable connection (or software that thinks it's unreliable) which reminded me that I had seen a protocol setting in NDI Tools Access Manager.  Checking there, I found the Receive mode was set to it was set to "Auto (Recommended)."  It seemed likely that "Reliable UDP" (which seems oxymoronic, but I digress) was being used by default, and testing seemed to confirm this as **choosing any other protocol (including regular old "unreliable" UDP) made the connection flapping and audio lagging disappear**.

While it feels to me that there might be a bug with Reliable UDP where it's detecting a connection drop when there isn't one, perhaps there is some additional network configuration requirement for Reliable UDP that just isn't well-documented?  I'm frustrated that the "Auto" setting apparently doesn't actually try to fallback to any other protocols automatically, even though I would think the flapping and high latency would logically suggest it would be worth trying another.  I'm unclear what's "auto"matic about this setting.

It would also help DistroAV users if the ability to change the Receive protocol was included in the plugin somewhere, so NDI tools aren't required.

Hope this helps someone!

**Reply by RayneYoruka** [reply]
I'll keep this in mind for when I see audio lagging issues in users.

Don't forget to contact NDI about this bug.
https://ndi.video/resources/get-in-touch/

**Reply by winterfrost1** [reply]
Thanks, RayneYoruka. I've sent a report to NDI.

**Reply by BolikexMachina** [reply]
> TLDR: Try using NDI Tools Access Manager to change the Receive protocol. (In my case, anything except "Reliable UDP.")
> 
> Just putting this out there in case anyone else is having similar issues: after fighting with OBS/NDI with audio cutout issues (usually logged as audio lagging, and then resetting at maximum buffers--see #875) for months, and finding no help online ("it's your network" was always the conclusion... but my network, like many others, shows no signs of issues with other applications, internet and local speed tests, file transfers, games, etc.) I finally found a solution.
> 
> Running NDI Analysis on my PCs showed the same audio connection flapping that you have reported. Immediately following the flapping, the next audio latency times were ~200ms higher than they were before. Every time the flapping occurs, the lag increases by ~200ms and this eventually causes the connection to lag out. I had tried multiple PCs, and went so far as to purchase a new PC and new core switch, but all systems (Windows 11) on both switches had the same issue.
> 
> Seeing the connection flapping suggested an unreliable connection (or software that thinks it's unreliable) which reminded me that I had seen a protocol setting in NDI Tools Access Manager. Checking there, I found the Receive mode was set to it was set to "Auto (Recommended)." It seemed likely that "Reliable UDP" (which seems oxymoronic, but I digress) was being used by default, and testing seemed to confirm this as **choosing any other protocol (including regular old "unreliable" UDP) made the connection flapping and audio lagging disappear**.
> 
> While it feels to me that there might be a bug with Reliable UDP where it's detecting a connection drop when there isn't one, perhaps there is some additional network configuration requirement for Reliable UDP that just isn't well-documented? I'm frustrated that the "Auto" setting apparently doesn't actually try to fallback to any other protocols automatically, even though I would think the flapping and high latency would logically suggest it would be worth trying another. I'm unclear what's "auto"matic about this setting.
> 
> It would also help DistroAV users if the ability to change the Receive protocol was included in the plugin somewhere, so NDI tools aren't required.
> 
> Hope this helps someone!


> TLDR: Try using NDI Tools Access Manager to change the Receive protocol. (In my case, anything except "Reliable UDP.")
> 
> Just putting this out there in case anyone else is having similar issues: after fighting with OBS/NDI with audio cutout issues (usually logged as audio lagging, and then resetting at maximum buffers--see #875) for months, and finding no help online ("it's your network" was always the conclusion... but my network, like many others, shows no signs of issues with other applications, internet and local speed tests, file transfers, games, etc.) I finally found a solution.
> 
> Running NDI Analysis on my PCs showed the same audio connection flapping that you have reported. Immediately following the flapping, the next audio latency times were ~200ms higher than they were before. Every time the flapping occurs, the lag increases by ~200ms and this eventually causes the connection to lag out. I had tried multiple PCs, and went so far as to purchase a new PC and new core switch, but all systems (Windows 11) on both switches had the same issue.
> 
> Seeing the connection flapping suggested an unreliable connection (or software that thinks it's unreliable) which reminded me that I had seen a protocol setting in NDI Tools Access Manager. Checking there, I found the Receive mode was set to it was set to "Auto (Recommended)." It seemed likely that "Reliable UDP" (which seems oxymoronic, but I digress) was being used by default, and testing seemed to confirm this as **choosing any other protocol (including regular old "unreliable" UDP) made the connection flapping and audio lagging disappear**.
> 
> While it feels to me that there might be a bug with Reliable UDP where it's detecting a connection drop when there isn't one, perhaps there is some additional network configuration requirement for Reliable UDP that just isn't well-documented? I'm frustrated that the "Auto" setting apparently doesn't actually try to fallback to any other protocols automatically, even though I would think the flapping and high latency would logically suggest it would be worth trying another. I'm unclear what's "auto"matic about this setting.
> 
> It would also help DistroAV users if the ability to change the Receive protocol was included in the plugin somewhere, so NDI tools aren't required.
> 
> Hope this helps someone!


THANKYOUUUUUUUUU this helped me alot, in my case SINGLE-TCP was a solution, MANY MANY THANKS

**Reply by djungowski** [reply]
> TLDR: Try using NDI Tools Access Manager to change the Receive protocol. (In my case, anything except "Reliable UDP.")

> THANKYOUUUUUUUUU this helped me alot, in my case SINGLE-TCP was a solution, MANY MANY THANKS

OMG, thank you guys. You just saved me from going insane (Single-TCP was also the right solution for me)

**Reply by RayneYoruka** [reply]
@djungowski @BolikexMachina Do you two happen to have the latest updates..?

In Windows 10: KB5063709.
In Windows 11: KB5063878.

**Reply by djungowski** [reply]
@RayneYoruka I do. I have also already seen articles that confirm that it's in regard to this update
